providers/aws/resources/aws_account.go

```diff
@@ -52,3 +51,50 @@ func (a *mqlAwsAccount) organization() (*mqlAwsOrganization, error) {
 		})
 	return res.(*mqlAwsOrganization), err
 }
+
+func (a *mqlAwsOrganization) accounts() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	client := conn.Organizations("") // no region for orgs, use configured region
+
+	orgAccounts, err := client.ListAccounts(context.TODO(), &organizations.ListAccountsInput{})
+	if err != nil {
+		return nil, err
+	}
+	accounts := []interface{}{}
+	for i := range orgAccounts.Accounts {
+		account := orgAccounts.Accounts[i]
+		res, err := CreateResource(a.MqlRuntime, "aws.account",
+			map[string]*llx.RawData{
+				"id": llx.StringDataPtr(account.Id),
+			})
+		if err != nil {
+			return nil, err
+		}
+		accounts = append(accounts, res.(*mqlAwsAccount))
+	}
+	return accounts, nil
+}
+
+func initAwsAccount(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if len(args) >= 2 {
+		return args, nil, nil
+	}
+	if len(args) == 0 {
+		if ids := getAssetIdentifier(runtime); ids != nil {
+			id := strings.TrimPrefix(ids.arn, "arn:aws:sts::")
+			args["id"] = llx.StringData(id)
+		}
+	}
+	if args["id"] == nil {
+		return args, nil, errors.New("nil account id")
```

Comment:
nit: I'm not sure users understand "nil", when they see this error. Just use "no" instead, or "no ... specified"? Or ...

Reply:
true true, ill change that